web/packages/quick-editor/src/index.ts

```diff
@@ -0,0 +1,78 @@
+export type GravatarQuickEditorOptions = {
+	avatarSelector?: string;
+	scope?: string[];
+}
+
+export default class GravatarQuickEditor {
+	_triggerElement: HTMLElement;
+	_avatarElements: NodeListOf<HTMLImageElement>;
+	_email: string;
+	_scope: string[];
+
+	constructor( triggerSelector: string, email: string, options: GravatarQuickEditorOptions ) {
+		this._triggerElement = document.querySelector( triggerSelector );
+		this._email = email;
+
+		if ( ! this._triggerElement ) {
+			console.error( 'Gravatar Quick Editor: Trigger element not found' );
+			return;
+		}
+
+		if ( ! this._email ) {
+			console.error( 'Gravatar Quick Editor: Email not provided' );
+			return;
+		}
+
+		if ( options.scope ) {
+			this._scope = options.scope;
+		}
+
+		if ( options.avatarSelector ) {
+			this._avatarElements = document.querySelectorAll<HTMLImageElement>( options.avatarSelector );
+		}
+
+		this.setupEvents();
+	}
+
+	setupEvents() {
+		this._triggerElement.addEventListener( 'click', () => {
+			this.showPopup()
+		} );
+
+		window.addEventListener( 'message', ( event ) => {
+			if ( ! event.origin.match( /https:\/\/([a-z]{2,3}\.)?gravatar.com/ ) ) {
```

Comment:
Does it works for `https://zh-tw.gravatar.com`?

Reply:
I didn't know there was a local string like that 😆 
I've adjusted it.